spinalcordtoolbox/reports/qc.py

```diff
@@ -57,10 +58,54 @@ class QcImage:
         "#ff0000",  # Red         ( 5.25:1)
         "#50ff30",  # Green       (15.68:1)
         "#F749FD",  # Magenta     ( 7.32:1)
-    ] * 15
+    ]
     _seg_colormap = ["#4d0000", "#ff0000"]
     _ctl_colormap = ["#ff000099", '#ffff00']
 
+    def _assign_label_colors_by_groups(self, labels):
+        """
+        This function handles label colors when labels may not be in a single continuous group:
+
+            - Normal vertebral labels:  [2, 3, 4, 5, 6, ...]
+            - Edge case, TotalSegmentator labels: [31, 32, 33, 200, 201, 217, 218, 219]
+
+        We assume that the subgroups of labesl (1: [31, 32, 33, ...], 2: [200, 201], 3: [217, 218, 219, ...])
+        should each be assigned their own distinct colormap, as to
+        """
+        # Arrange colormaps for max contrast between colormaps, and max contrast between colors in colormaps
+        distinct_colormaps = ['Blues', 'Reds', 'Greens', 'Oranges', 'Purples']
+        colormap_sampling = [0.25, 0.5, 0.75, 0.5]  # light -> medium -> dark -> medium -> (repeat)
+
+        # Split labels into subgroups --> we split the groups wherever the difference between labels is > 1
+        start_end = [0, len(labels)]
+        for idx, (prev, curr) in enumerate(zip(labels, labels[1:]), start=1):
+            if curr - prev > 1:
+                start_end.insert(len(start_end)-1, idx)
```

Comment:
If you wait until after the loop to put `len(labels)` at the end of `start_end`, then this `start_end.insert` could be a slightly simpler `start_end.append`. (This is a super optional suggestion.)

Reply:
I think I originally had it as you suggest? But, when I was looking at the "append" version, it felt... unclear? So, I rewrote it this way in the hopes of making the "`start_end`" variable behavior more clear, since it's a bit of a weird variable name to grok, hehe. 

(With the insert version, by default, we start with one full "group" with `start = 0` and `end = len(labels)`. Then, whenever we encounter a new "break" in the group, we insert that index, such that we have two groups `(0, i)` and `(i, len(labels))`.)